### PR TITLE
Remove redundant stages in Jenkinsfile

### PR DESCRIPTION
**Pull request by willis7**
Remove unused code, use a convention for post stage messages. Rename
the Package stage. Put sbt on the path.

REG-427

### CODE REVIEW

**Comment by willis7** [comment]
Ready for review @ameenhaq @ranjit-das @awharris. This PR is currently blocking #56.

**Comment by willis7** [comment]
Comments from @ameenhaq via Slack

> So two things regarding your Jenkinsfile (got an email it was failing)
> 1) I think test stage could be merged into Static Analysis stage - just my opinion.
> 2) Your build is probably failing due to a kerberos ticket not being created. I would not recommend deploying the HBase artifact. If however you wish to I would firstly confer with my man Ranjit
> 
> He would probably advise you to deploy to dev for the HBase (as the user is active and thereby a ticket is current). Else (if you insist on a test HBase Load jar deployment) then you should generate a ticket - gimme a shout if you want more details on this.
> Being pedantic, but I would change ‘Deploy CF’ => ‘Deploy API CF’ and ‘Deploy HBase’ => ‘Deploy load (JAR) HBase’ - that way its clear to other users :slightly_smiling_face:

1) Good point I will provide a fixup for parallel test.
2) Agreed, using dev for now.
3) I'm going to keep the more general names for now as it improves reusability. Happy to change if another reviewer would like to see the change.

**Comment by willis7** [comment]
I reverted the parallelization of unit test. It appears scapegoat and the unit tests are competing for the same resource, which causes the unit test stage to fail. It's a worthwhile change to make, but not for this ticket.

File: Jenkinsfile [hunk 1]
```diff
@@ -41,95 +34,90 @@ pipeline {
...
                     version = '1.0.' + env.BUILD_NUMBER
                     currentBuild.displayName = version
-                    STAGE = "Checkout"
                 }
             }
         }
 
         stage('Build') {
             agent any
             steps{
```
Comment (ameenhaq) [inline Jenkinsfile]: environment{ STAGE = "Build" }

File: Jenkinsfile [hunk 2]
```diff
@@ -41,95 +34,90 @@ pipeline {
...
                 }
             }
         }
 
         stage('Static Analysis') {
             agent any
+            environment{ STAGE = "Static Analysis" }
             steps {
                 parallel (
                     "Scalastyle" : {
```
Comment (ameenhaq) [inline Jenkinsfile:90]: I assume IT tests will live here as well - although I don't think theres any for this project (which is another task in itself)

File: Jenkinsfile [hunk 3]
```diff
@@ -140,149 +128,92 @@ pipeline {
                 cp gitlab/dev/conf/* conf
```
Comment (ameenhaq) [inline Jenkinsfile:128]: 119-129 - don't think these are needed anymore as we don't use synthetic csv data anymore.

File: Jenkinsfile [hunk 4]
```diff
@@ -140,149 +128,92 @@ pipeline {
...
                 lock('PAYE Deployment Initiated') {
-                    colourText("info", "${env.DEPLOY_NAME}-${PAYE_TABLE}-${MODULE_NAME} deployment in progress")
+                    colourText("info", "${env.DEPLOY_TO}-${PAYE_TABLE}-${MODULE_NAME} deployment in progress")
                     deploy(PAYE_TABLE, false)
-                    colourText("success", "${env.DEPLOY_NAME}-${PAYE_TABLE}-${MODULE_NAME} Deployed.")
+                    colourText("success", "${env.DEPLOY_TO}-${PAYE_TABLE}-${MODULE_NAME} Deployed.")
                 }
-		        lock('Legal Unit Deployment Initiated') {
-                    colourText("info", "${env.DEPLOY_NAME}-${LEU_TABLE}-${MODULE_NAME} deployment in progress")
+                lock('Legal Unit Deployment Initiated') {
```
Comment (ameenhaq) [inline Jenkinsfile]: Legal Unit?

File: Jenkinsfile [hunk 5]
```diff
@@ -140,149 +128,92 @@ pipeline {
...
+            post {
+                success {
+                    colourText("info","Stage: ${env.STAGE} successful!")
                 }
-            }
-            steps {
-                script {
-                    env.NODE_STAGE = "Package and Push Artifact"
+                failure {
+                    colourText("warn","Stage: ${env.STAGE} failed!")
```
Comment (ameenhaq) [inline Jenkinsfile:174]: Maybe in the future we could capture which API fails here to make it easier to the user.

File: Jenkinsfile [hunk 6]
```diff
@@ -140,149 +128,92 @@ pipeline {
...
             steps {
-                script {
-                    STAGE = "Releases"
-                    currentTag = getLatestGitTag()
-                    colourText("info", "Found latest tag: ${currentTag}")
-                    newTag =  IncrementTag( currentTag, RELEASE_TYPE )
-                    colourText("info", "Generated new tag: ${newTag}")
-                    //push(newTag, currentTag)
-                }
+                sh "sbt 'set test in assembly := {}' assembly"
```
Comment (ameenhaq) [inline Jenkinsfile:187]: Ranjit needs to fix this. This is omitting the tests when assembling the JAR. The cause I believe was that it needed to execute something in the 'tmp' folder - sounds familiar I'm sure.

File: Jenkinsfile (hunk 2 above)
Comment (willis7) [reply Jenkinsfile:90]: I avoid that. If you run it after the faster unit and static analysis, you save CPU resource in the event one of them fails.

File: Jenkinsfile (hunk 4 above)
Comment (willis7) [reply Jenkinsfile]: @ranjit-das asked me to put it back in

File: Jenkinsfile (hunk 3 above)
Comment (willis7) [reply Jenkinsfile:128]: This task isn't about refactoring the pipeline entirely, I just need to remove references to branches. We've been scope creaping already - I would suggest raising a Jira if this isnt required.

File: Jenkinsfile (hunk 6 above)
Comment (willis7) [reply Jenkinsfile:187]: This is a pretty good optimization; we run `test` in the `Test` stage, we dont need to run it again here 👍

File: Jenkinsfile (hunk 5 above)
Comment (willis7) [reply Jenkinsfile:174]: We would have to script that behaviour. Lets catch it in a Jira.

File: Jenkinsfile (hunk 6 above)
Comment (ameenhaq) [reply Jenkinsfile:187]: Good point

File: Jenkinsfile (hunk 4 above)
Comment (ranjit-das) [reply Jenkinsfile]: @willis7 I have realised that I misundrestood it with LEU load available in GitLab. So could you please remove it from here. Sorry for the confusion.

**Pull request merged**